app/models/workflow/instance/step.rb

```diff
@@ -9,8 +9,8 @@ class Instance::Step < ApplicationRecord
     belongs_to :selected_option, class_name: 'Workflow::DecisionOption', optional: true
 
     before_create :set_position
-    after_save :update_completed_counter_cache
-    after_destroy :update_completed_counter_cache
+    after_save :update_process
+    after_destroy :update_completed_counter_cache, :update_process
```

Comment:
I think there's a simpler way to do this but haven't done it in a while.
`belongs_to :article, counter_cache: true`
https://blog.appsignal.com/2018/06/19/activerecords-counter-cache.html

Reply:
according to the article, there's no built in way to do a scoped counter cache, unless you use a gem.

Reply:
Think I'm going to keep it this way because of sequencing of `update_process`. Need to make sure the `update_process_completed_counter_cache` is run before `update_process_completion_status`.

Reply:
ah thx sounds good